logicrunner/logicrunner.go

```diff
@@ -337,6 +356,17 @@ func (lr *LogicRunner) pulse(ctx context.Context) *insolar.Pulse {
 	return &p
 }
 
+func contextWithServiceData(ctx context.Context, data message.ServiceData) context.Context {
```

Comment:
why do you need this function? what do you want to keep in context from old context?

Reply:
flow.Pulse() won't work otherwise.

Reply:
this is bad :(